tests/test_migrate_config.py

```
@@ -0,0 +1,423 @@
+        assert cfg.github_repo == "user/repo"
+
+    def test_no_migration_on_db_read_failure(self, tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> None:
+        """A DB read failure must not trigger auto-migration (fail-closed)."""
+        monkeypatch.delenv("SOVA_DATABASE_URL", raising=False)
+        db_path = _create_db(tmp_path)
+        _seed_db_settings(db_path, {"github_repo": "db/repo"})
+        _write_toml(tmp_path, '[project]\ngithub_repo = "toml/repo"\n')
+
+        from sova.config import db_loader
+
+        def _failing_load(pd: Any) -> None:
```

Ruff (0.16.2)

[warning] Unused function argument: `pd`

(ARG001)